Reimagines Station Orbiting: New local system - Auri

TsunamiAnt opened the pull request:
# TLDR & Gameplay:
- There are now orbital thrusters and a console to control them.
- The station now orbits at a specific altitude.
- The crew needs to ensure there is enough fuel to run the thrusters.
- If the fuel runs out and the station is not prepared, station loss is possible.
- You can buy more fuel in cargo.
- This lays the first groundwork for our new plasma-focused game design.
  - The station starts without any gas in the atmos tanks, encouraging mining efforts

# The orbital map has been completely reworked

- Now a whole solar system is present.
- Orbiting the star in the middle(Auri) is:
- Cinis and it's asteroid belt (Lavaland)
- The station orbits close to cinis.
- Neo has been created. More of a collection of asteroids than a proper moon, this baby (provisionally) holds CC and nukies.
- Further out, Thetis and Thallos orbit.
- And even further, triea.
- When playing on echostation, located on thallos, thallos is moved into orbit of lavaland for convenience. This is not canon.

# Station Orbital Mechanics
- The station now has an orbital altitude mechanic. You can control at what altitude the station is orbiting.
- Added a computer to control the station's thrusters, letting crew adjust the station's position.
- Added orbital thrusters and thruster fuel (purchaseable from cargo) to power the system.
- Different orbital altitudes provide different effects to the station and its crew.
- Many different systems interact with orbital altitude, examples include the gateway, cargo shuttles, and solar power.

# New Space Background System
- Parallax has been entirely removed.
- Replaced with a new space background system that dynamically changes based on station altitude.
- Starlight now adapts to the altitude of the station.
- All of this respects the Z-level of the viewer of course.

## Why It's Good For The Game

Design goals demands this.
It is very cool.
It brings us in line with proposed lore.
The visuals are also very cool.
The necessity for a specific resource adds tension and requires prioritization- you've read the design pillars. It's this. This entire thing has been floated around for like months now, we all know about it.

## Testing Photographs and Procedure
<details>
<summary>Screenshots&Videos</summary>

Altered starmap:
<img width="1968" height="1266" alt="image" src="https://github.com/user-attachments/assets/8885099b-c0e0-4800-903d-35b2788e35bb" />

fuel
<img width="927" height="464" alt="image" src="https://github.com/user-attachments/assets/51f7e448-b65e-4dae-985b-d8ec18632a3e" />

subsystems
<img width="532" height="255" alt="image" src="https://github.com/user-attachments/assets/580ef93a-2f6b-4a72-b099-4f918e66dca7" />

warnings
<img width="1061" height="140" alt="image" src="https://github.com/user-attachments/assets/ad0befc1-786a-4164-85f4-9e187f2df3db" />

thrusters
<img width="493" height="896" alt="image" src="https://github.com/user-attachments/assets/2b47c905-8723-4e08-a179-dc614e8c3fdc" />

new background 1
<img width="1449" height="1269" alt="image" src="https://github.com/user-attachments/assets/06b73771-4324-4ce3-9649-41a5c41b1e21" />

inside the atmosphere
<img width="1231" height="1080" alt="image" src="https://github.com/user-attachments/assets/ab1d9526-73da-4198-ad94-996d04f7484e" />

burning up
<img width="1079" height="1061" alt="image" src="https://github.com/user-attachments/assets/efda0949-b5ed-4d65-a984-0adbca68cda7" />

re-entry damage:
<img width="958" height="1012" alt="image" src="https://github.com/user-attachments/assets/9eae01b6-a15f-45aa-9770-e5d4fd2d765b" />

<img width="990" height="563" alt="image" src="https://github.com/user-attachments/assets/6fa66d79-e873-4460-851e-2e51e9261dfb" />

controller
<img width="945" height="820" alt="image" src="https://github.com/user-attachments/assets/ddfad991-fe81-44e9-adf7-ab26cdfec8b5" />

re-entry effects and such do not influence other z-levels:
<img width="1192" height="1131" alt="image" src="https://github.com/user-attachments/assets/306329fc-f7d3-4b5a-9953-fb5035536a42" />

Travel works fine:
<img width="1107" height="965" alt="image" src="https://github.com/user-attachments/assets/1e3988ee-77b9-45c5-9879-203c3dec5a01" />

</details>

## Changelog
:cl: TsunamiAnt, BriggsIDP
add: Added orbital altitude to the station
add: Added a computer to control the station's thrusters
add: Added orbital thrusters
add: Added orbital thruster fuel, purchaseable in cargo
add: Added different effects depending on orbital altitude
del: Removed parallax
add: Added a new space background system based on station altitude
add: Add a system where starlight adapts to the altitude of the station/the person viewing it
tweak: tweaked the orbital map to reflect the one proposed by lore:
add: Added Auri Geminae, the main system star.
tweak: Renamed lavaland to Cinis.
tweak: Moved the station orbit much much closer to it.
tweak: Renamed phobos to Thetis, a large gas giant around which the asteroids are orbiting.
tweak: If on echo, the station now orbits Thetis instead, as 5132-E or whatever it's called. The jungle moon.
add: Added triea on the outskirts of the system, a small icy rock.
tweak: Moved the signals and beacons into orbit around the sun, between the orbits of thetis and triea.
balance: removed plasma from the roundstart atmos tank
fix: fixed a few bugs all around related systems to make them work
sound: added a sound to the AGR retracting
tweak: modified agr functioning to account for orbital distance
image: added some self-made art for the new background
map: added/modified/removed map content
spellcheck: fixed a few typos
config: In the process of removing parallax, I had to remove some database touching/adjacent code. Please correct any mistakes.
admin: In the process of removing parallax, I had to remove some database touching/adjacent code. Please correct any mistakes.
server: In the process of removing parallax, I had to remove some database touching/adjacent code. Please correct any mistakes.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

TsunamiAnt committed:
Adds orbital height subsystem and gas rig overhaul

Introduces an orbital height subsystem that simulates changes in station altitude, which affects radiation levels and structural integrity.

Overhauls the gas rig mechanics, transitioning from depth-based to extension-based calculations.  This involves modifying shield calculations, power consumption, gas production, and repair mechanics.  Prevents rig activation when nozzle is unsafe.

Refactors code for clarity and includes associated UI updates and sounds.

TsunamiAnt committed:
Improves Atmos gas rig nozzle indicator rendering

Reorders the rendering of the station blip and nozzle extension indicators to ensure they are drawn on top of the tickmarks, enhancing visibility.

TsunamiAnt committed:
Adds orbital altitude subsystem

Implements a new subsystem to simulate and manage the station's orbital altitude.

This subsystem introduces dynamic altitude changes, taking into account factors such as decay rate, thrust, and atmospheric resistance.
It also handles critical orbit conditions, triggering alerts and initiating a destruction sequence if the station remains in low orbit for too long.
The system includes atmospheric drag effects with associated visual and audio feedback.
A newsfeed channel is created for periodic automated updates on station orbital status.

TsunamiAnt committed:
Balances atmospheric drag meteor impact

Reduces the frequency of heavy atmospheric drag meteors
by implementing a cooldown. Increases the damage of
heavy meteors and decreases the damage of light meteors.

TsunamiAnt committed:
Hides atmospheric drag meteor effect

Sets the atmospheric drag meteor effect to be invisible by default.
This change prevents the effect from being visible unless specifically enabled for debugging.
Also re-enables the removal of the atmospheric drag meteor from the augury list.

TsunamiAnt committed:
Implements orbital decay and station destruction.

Adds a system to simulate orbital decay, atmospheric drag,
and eventual station destruction if the station falls too low.

This system introduces altitude thresholds, decay rates,
thrust mechanics, and atmospheric resistance to create a dynamic
orbital environment. It incorporates visual effects and
announcements to provide player feedback and warning of
critical situations, and performs station destruction
if the station is not recovered from a critical state.

TsunamiAnt committed:
Decouples re-entry lighting from orbital altitude.

Moves the atmospheric re-entry lighting effects to its own subsystem.

This improves modularity and allows for independent control and easier
management of visual effects related to re-entry.

Orbital altitude now references the new subsystem for activating
and deactivating the re-entry lighting effects.

TsunamiAnt committed:
Implements reentry erosion effect.

Adds a reentry erosion subsystem that damages stations
during orbital descent. The intensity of the effect scales
with altitude. The system reads the map configuration to
determine the correct direction for the effect. Also increases
the orbital height and adjusts atmospheric drag cooldown.

TsunamiAnt committed:
Adds orbital height control console

Adds a new computer console for monitoring and controlling the station's orbital height.

This console allows for future implementation of features related to station positioning systems.

TsunamiAnt committed:
Replaces Orbital Height Computer with TV

Replaces the orbital height control computer in the bridge map with a wooden TV.
Also swaps a snack spawner for a glass table in the same area.
Fixes some minor typo errors in map definition.

TsunamiAnt committed:
Adds low altitude warning system

Introduces a low altitude warning system that triggers when the station dips below 95km.

This enhancement provides an early warning to players, allowing them time to react and prevent critical orbital decay.
The system includes both visual and audible alerts and integrates with existing atmospheric effects for increased immersion.

The system will also clear the low altitude warning flag when the station is safely above 95km.

TsunamiAnt committed:
Adds orbital height control console

Introduces an orbital height control console for monitoring and adjusting the station's altitude.

This includes:
- A new computer object with UI for displaying orbital data (altitude, decay rate, velocity index, atmospheric resistance, thrust level).
- Visualization of orbital bands and the station's current altitude.
- Thrust control functionality to adjust orbital parameters.

TsunamiAnt committed:
Adds orbital height control system

Adds an orbital height control system to the computer,
allowing for manual thrust control and altitude hold functionality.

Includes a planet visualization with orbital bands.
Also displays warnings for dangerous orbital conditions.

TsunamiAnt committed:
Adds orbital thruster and altitude control system

This introduces a new orbital thruster system to control the station's altitude.

Adds orbital thruster objects that can be placed on the station.
Introduces a control computer that allows players to manually set the thrust level,
or enable altitude hold to automatically maintain a target altitude.

Improves orbital altitude calculations, taking into account thruster force.

TsunamiAnt committed:
Improves orbital thruster functionality

Adds a looping sound effect and visual glow to orbital thruster nozzles.

Adjusts thrust damage application by casting a ray for multiple tiles and applying damage falloff.

Modifies propellant consumption logic to incrementally adjust thrust levels.

Replaces some thruster dummy objects on the map with functional versions and warning signs.

TsunamiAnt committed:
Adds orbital thrusters and radiation band

Adds orbital thrusters and orbital radiation band subsystems and components for orbital mechanics and associated radiation effects.

The orbital thruster system uses formulated hydrogen fuel to allow for orbital maneuvers. A gas monitor is used for hydrogen fuel. A nozzle object spawns the thruster visual effect based on engine thrust.

Implements an orbital radiation band subsystem that applies radiation based on altitude, increasing with altitude and radiation intensity. Includes radiation immunity and protection based on station location. Radiation band starts at 120km and becomes critical at 130km.

Updates re-entry erosion priority and adds orbital altitude checks.

TsunamiAnt committed:
clean up

TsunamiAnt committed:
Removes hysteresis from fire effect cutoff

Removes the hysteresis from the atmospheric fire effect cutoff logic.

This simplifies the code and prevents potential issues related to the fire effect not being consistently disabled at the intended altitude.

TsunamiAnt committed:
Parallax gone. Space background welcome.

TsunamiAnt committed:
Massive cleanup after code-dead-end

Implements a new orbital visuals system that uses altitude-based layers for a more dynamic and visually appealing space environment.

Replaces the previous system with a layered approach, where different visual elements (space background, stars, atmosphere, fire, celestial bodies) are displayed and animated based on the station's current orbital altitude.

The new system provides a more immersive experience as the station ascends and descends through the atmosphere.

TsunamiAnt committed:
Refactors orbital visuals and altitude system.

Refactors the orbital altitude and visual systems for improved accuracy and flexibility.

Now correctly uses highest altitude for non-station Z levels.

Reverses the reentry direction.

Fixes starlight color control and ensures starlight color overrides are properly managed by the orbital visuals subsystem.

Sets altitude hold to true by default.

TsunamiAnt committed:
Uses absolute thrust for damage calculation.

Ensures that thrust damage and visual representation are always positive, regardless of thrust direction.

This prevents potential issues where negative thrust values could lead to unexpected behavior in damage calculations and particle effects.

TsunamiAnt committed:
Touch starlight plane master to allow for per-client starlight colors.

TsunamiAnt committed:
Adds missing hydrogen canister to engi start

Ensures engine has the necessary hydrogen canister on map load for the orbitalheight branch.

TsunamiAnt committed:
Clarifies starlight override system

Improves clarity on how starlight color overrides function between the natural light cycle, aurora event, and orbital visuals subsystems.

Adds comments explaining the purpose of set_starlight_colour and set_orbital_starlight_colour procs, as well as when and why to use the orbital visuals override.

Ensures starlight override is enabled when the natural light cycle is active.

TsunamiAnt committed:
Adds orbital height control research and supplies

Adds research for orbital height control.

Adds supply packs for hydrogen fuel canisters to assist with orbital maneuvering. Single, double, and triple packs are available with increasing discounts.

TsunamiAnt committed:
Adds non-oxygenated fire creation and adds gas reaction to fhf

Implements hydrogen fuel as a new flammable gas.

This change introduces hydrogen fuel as a new gas type, defining its properties and behavior within the atmospherics system.
It can be used as a propellant for orbital thrusters, and is mildly toxic when inhaled.
The combustion of hydrogen fuel produces water vapor.

TsunamiAnt committed:
Fix orbital map

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
commit message

TsunamiAnt committed:
Prevents errors with no thrusters.

Prevents division by zero when no thrusters are present.

Undefines orbital state macros to avoid potential conflicts.

TsunamiAnt committed:
Woops i left one!

TsunamiAnt committed:
Removes outdated radiation band comments.

Removes outdated comments pertaining to the orbital radiation band subsystem, as the system has been overhauled.

TsunamiAnt committed:
Clarifies hydrogen fuel combustion description.

Updates the description of hydrogen fuel combustion to
accurately reflect that it only involves hydrogen fuel.

TsunamiAnt committed:
Reduces space station mass.

Reduces the mass of the space station to prevent
unrealistic orbital behavior.

TsunamiAnt committed:
Reworks orbital map and planetary setup

Refactors the orbital map generation to remove hardcoded planet instances and improve flexibility.

Introduces a more generalized approach to planet configuration, reading planetary data from map JSON files.
This allows for dynamically defined planetary systems based on the loaded map, simplifying the addition of new planets and orbital bodies.

Replaces specific planet objects (Thetis, Triea) with a generic echo planet and custom lavaland planet.
Increases initial asteroid count in the orbital map.

TsunamiAnt committed:
Removes unused orbital altitude definition file.

Removes a now obsolete definition file. The content is no longer relevant to the project.

TsunamiAnt committed:
Makes lavaland a static orbital object.

Prevents lavaland from moving during supercruise.
This ensures its position remains consistent for
docking and other interactions.

TsunamiAnt committed:
Improves orbital reentry effects and fixes

Changes the atmosphere overlay to blend more effectively.

Prevents errors caused by destroyed objects during reentry
by checking object validity after fire interaction.

TsunamiAnt committed:
Improves orbital map object visibility.

Ensures orbital map objects remain visible even when partially off-screen by accounting for their radius in bounds checks.

Adds fallback logic for tracking the shuttle's location on the orbital map, switching to the docked location's name when the shuttle is docked.
This is achieved by linking the console's z-level to an orbital object.

Initial zoom scale is also adjusted.

TsunamiAnt committed:
Adds tutorial notes to orbital controls

Adds sticky notes to the orbital height control and fuel supply computer to provide players with more information on how to operate them.

Adds some extra flavor text to the nitrium fuel description.

Prevents thrust particles from displaying if the thruster backend does not exist, or it is out of fuel.

PowerfulBacon commented on code/controllers/subsystem/station_orbital/orbital_reentry_erosion.dm:
```diff
@@ -0,0 +1,259 @@
+	return FALSE
+
+
+/datum/controller/subsystem/orbital_reentry_erosion/fire(resumed)
```
The code originally in here that handled optimising the work queue was removed, and now has_real_contents gets run about 7000 times every second. It needs to return. If you need help with this then message me on discord since I can help with anything overly technical. To get what you want you need two things:
- An array which holds the coordinate of the tile currently being burnt.
- An array which holds the coordinate of the tile currently being inspected.

If the burning tile is not null, burn it.
If the burning tile is set to null, repeat the following 5 times
- If the inspected tile is equal to the burning tile, reset the inspected tile back to the start of the z-level.
- Increment the inspected tile
- If the inspected tile comes across something that is real, set the burning tile to that tile and break. Reset the inspected tile back to the start.

Using this algorithm, your worst case scenario is 1275 calls to has_real_contents per tick. For most stations this should be called ~655 times which makes it about 90% faster while still allowing the burning to travel backwards if things start to get rebuilt.

You really don't want to be recalculating the tile that you are burning every second by scanning the entire z-level since that is going to result in unnecessary CPU work.

TsunamiAnt replied:
I know, I am not doing that.

PowerfulBacon commented on code/controllers/subsystem/orbital_reentry_erosion.dm:
```diff
@@ -0,0 +1,259 @@
+		if (!isobserver(thing))
+			return TRUE
+
+	return FALSE
```
Remove is_real as proc call overhead starts to become an issue here and just inline it. has_real_contents might be beneficial to remove too

Rukofamicom replied:
Again no not resolve comments that weren't addressed

TsunamiAnt replied:
Ruko this entire subsystem was deprecated and split into two separate ones

TsunamiAnt replied:
No longer required due to changes in how we handle it. I will still consider doing it, however.

TsunamiAnt replied:
Is_valid_hit has been inlined, has_real_contents will not be inlined because we only call it when we actually land on a tile with valid contents, so overhead is negligible.

PowerfulBacon commented on code/controllers/subsystem/orbital_reentry_erosion.dm:
```diff
@@ -0,0 +1,259 @@
+		return FALSE
+
+	var/has_contents = FALSE
+	for (var/atom/movable/thing in tile)
```
Use an as anything loop here

Rukofamicom replied:
Do not resolve comments that weren't addressed.

TsunamiAnt replied:
Noted.

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
wah

TsunamiAnt committed:
Adds orbital thruster status to OHC comp

Adds orbital thruster component status panel to the orbital height control computer UI to show status of connected thrusters.

Corrects typos.

Refactors map to include more space and remove some old code.

TsunamiAnt committed:
Sloppily slops the Slop-UI into less of a slop, it's acceptable-ish, now

TsunamiAnt committed:
it just did this on it's own, please see last pr title

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
BEGONE, SLOP

TsunamiAnt committed:
Updates orbital altitude announcements to minor alerts

Replaces priority_announce calls with minor_announce across the orbital altitude subsystem to reduce announcement prominence. Critical thresholds retain their alert status through the alert parameter.

TsunamiAnt committed:
Optimization and cleanup

TsunamiAnt committed:
Refactor orbital reentry into separate scanning and erosion subsystems

Separates the expensive raycasting logic from the damage application process to improve performance. The scanning subsystem now populates a shared list of target turfs that the erosion subsystem consumes, with both systems updated to be tick-aware and resumable. Centralized activation and deactivation are now handled by the orbital altitude subsystem.

TsunamiAnt committed:
Persist orbital reentry fire effects across subsystem cycles

Reuses existing fire effects instead of recreating them every cycle to prevent visual flickering and reduce object churn. Stale effects are now cleared based on updated scanning results. Additionally, caches the damage multiplier to ensure consistency when processing is resumed across ticks.

TsunamiAnt committed:
Run prettier through hell and back only for it to fix a single fucking -

TsunamiAnt committed:
Map config can be stale, add a fallback to read from file just in case.

TsunamiAnt committed:
Why are we reading from a list that does not exist?

TsunamiAnt committed:
Fixes issue where blendRGB at amount "1" exactly just loops to the start

We keep loop for hsl because that stuff's circular by design.

TsunamiAnt committed:
Optimize orbital starlight updates and prevent redundant animation calls

Throttles the orbital visuals subsystem fire rate and implements a caching wrapper for starlight color changes. This prevents unnecessary animate() calls when the target color remains unchanged, reducing overhead during atmospheric transitions while maintaining smooth interpolation for re-entry effects.

TsunamiAnt committed:
Adjust orbital flicker frequency, improve subsystem timing consistency

Switches the orbital visuals subsystem to SS_KEEP_TIMING and removes the background flag to ensure a more reliable fire rate for visual effects. The flicker probability has been updated to a more aggressive curve based on transition progress, and admin debug messaging has been added for tuning the effect.

TsunamiAnt committed:
Remove debug admin messaging from orbital visuals subsystem

Removes the temporary debug variable and admin notification used for tuning the orbital transition flicker frequency.

TsunamiAnt committed:
Refactor orbital re-entry erosion scaling and damage types

Simplifies the altitude thresholds for re-entry damage by removing the 80km bracket and capping maximum intensity at 85km. Re-entry damage now uses quadratic scaling for a less aggressive ramp-up and utilizes the MELEE damage flag to bypass fire armor, ensuring atmospheric friction remains a significant threat regardless of standard fire protections.

TsunamiAnt committed:
Move station into a more reasonable orbit

TsunamiAnt committed:
Refactor orbital radiation activation and intensity scaling

Shifts control of the radiation band subsystem to the orbital altitude controller, allowing it to be enabled or disabled based on state changes rather than polling altitude every fire. The radiation intensity logic is simplified to use discrete steps for the 120km and 130km thresholds.

TsunamiAnt committed:
Ensure orbital radiation band activates at critical altitude thresholds

Adds an explicit activation for the radiation band subsystem when entering the high altitude critical threshold. This ensures the radiation effects are enabled even if the station transitions directly into critical altitudes, bypassing the standard high altitude warning state.

TsunamiAnt committed:
fire

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

mrmanlikesbt commented on code/__HELPERS/icons.dm:
```diff
@@ -311,7 +311,7 @@ world
  */
 /proc/BlendRGB(rgb1, rgb2, amount)
-	return rgb_gradient(amount, 0, rgb1, 1, rgb2, "loop")
+	return rgb_gradient(amount, 0, rgb1, 1, rgb2)
```
why remove the "loop"?

TsunamiAnt replied:
Because it forced any entry with 1 to automatically loop around and be 0.

Removing loop does not alter it's functionality anywhere else it's used from what i can tell, and it clears up a strange case where, when trying to form a gradient, the utter right edge is the same color as the leftmost edge.

I used it in the starlight thing to blend from red to blue for example, and when you moved down to the lowest degree where the result of the math would always be 1 exactly, it would just flip back from blue(correct) to red.

mrmanlikesbt commented on code/_onclick/hud/orbital_visuals.dm:
```diff
@@ -0,0 +1,300 @@
+
+/atom/movable/screen/orbital_layer/Initialize(mapload, hud_owner)
+	. = ..()
+	// Don't call create_tiled_overlays here - it will be called manually after creation
```
```suggestion
```

mrmanlikesbt commented on code/_onclick/hud/orbital_visuals.dm:
```diff
@@ -0,0 +1,300 @@
+	if(!view)
+		view = world.view
+
+	// Mark this layer as tiled
```
```suggestion
/atom/movable/screen/orbital_layer/proc/create_tiled_overlays(view = world.view)
	// Mark this layer as tiled
```

mrmanlikesbt commented on code/_onclick/hud/orbital_visuals.dm:
```diff
@@ -0,0 +1,300 @@
+	// Calculate how many tiles we need in each direction based on view size
+	var/list/viewscales = getviewsize(view)
+	var/countx = CEILING((viewscales[1]/2)/(480/world.icon_size), 1) + 1
+	var/county = CEILING((viewscales[2]/2)/(480/world.icon_size), 1) + 1
```
```suggestion
	var/countx = CEILING((viewscales[1]/2)/(480/ICON_SIZE_X), 1) + 1
	var/county = CEILING((viewscales[2]/2)/(480/ICON_SIZE_Y), 1) + 1
```

mrmanlikesbt commented on code/modules/shuttle/on_move.dm:
```diff
@@ -182,14 +182,12 @@ All ShuttleMove procs go here
-/area/proc/afterShuttleMove(new_parallax_dir)
-	parallax_movedir = new_parallax_dir
+/area/proc/afterShuttleMove()
 	return TRUE
```
this doesn't seem to be overridden anywhere. remove this proc

mrmanlikesbt commented on config/performance.txt:
```diff
@@ -1,7 +0,0 @@
-
-## Set this to disable parallax updates after this amount of players have joined the server
-## This will cause the parallax to be a static background.
-PARALLAX_DISABLE_POP 60
-## The population at which ghosts should stop seeing parallax updates.
-## Alive players will still receieve them.
-PARALLAX_GHOST_DISABLE_POP 40
```
remove the whole file- not just all of the lines.

mrmanlikesbt commented on code/game/machinery/computer/orbital_height_control.dm:
```diff
@@ -0,0 +1,175 @@
+		))
+	data["thrusters"] = thrusters
+
+	return data
```
check if anything can be moved to `ui_static_data` here. `data["orbital_decay"]` is probably one

mrmanlikesbt commented on code/game/machinery/computer/orbital_height_control.dm:
```diff
@@ -0,0 +1,175 @@
+
+/obj/machinery/computer/orbital_height_control/ui_state(mob/user)
+	return GLOB.default_state
+
```
```suggestion
```

mrmanlikesbt commented on code/game/machinery/computer/orbital_height_control.dm:
```diff
@@ -0,0 +1,175 @@
+			altitude_hold_enabled = !altitude_hold_enabled
+			. = TRUE
+
+	return TRUE
```
only call a UI update if you need a UI update
```suggestion
	switch(action)
		if("increase_thrust")
			// Increase thrust by 1, max 20
			set_thrust = clamp(set_thrust + 1, -20, 20)
			return TRUE
		if("decrease_thrust")
			// Decrease thrust by 1, min -20
			set_thrust = clamp(set_thrust - 1, -20, 20)
			return TRUE
		if("set_altitude_hold_target")
			// Set the target altitude for altitude hold system
			var/target = text2num(params["target"])
			if(isnull(target))
				return FALSE
			altitude_hold_target = clamp(target, 80000, 140000)  // 80km to 140km in meters
			return TRUE
		if("toggle_altitude_hold")
			// Toggle altitude hold system on/off
			altitude_hold_enabled = !altitude_hold_enabled
			return TRUE
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+
+/obj/effect/meteor/atmospheric_drag/CanPass(atom/movable/mover, border_dir)
+	// Always let mobs pass through
+	if(isliving(mover) || ismob(mover))
```
```suggestion
	if(ismob(mover))
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+
+/obj/effect/meteor/atmospheric_drag/CanPassThrough(atom/blocker, turf/target, blocker_dir)
+	// Always pass through mobs
+	if(isliving(blocker) || ismob(blocker))
```
```suggestion
	if(ismob(blocker))
```

mrmanlikesbt commented on code/controllers/subsystem/station_orbital/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+	hits--
+	if(hits <= 0)
+		// Play creaking sound for atmosphere
+		playsound(src.loc, pick('sound/effects/creak1.ogg', 'sound/effects/creak2.ogg'), 80, TRUE, 300, falloff_distance = 300)
```
```suggestion
		playsound(src, pick('sound/effects/creak1.ogg', 'sound/effects/creak2.ogg'), 80, TRUE, 300, falloff_distance = 300)
```

TsunamiAnt replied:
I had that first, it will behave weirdly with walls and not transmit the sound reliably.

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+
+	// Damage structures and machinery
+	if(A.density)
+		if(istype(A, /obj/structure) || istype(A, /obj/machinery))
```
```suggestion
		if(isstructure(A) || ismachinery(A))
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+			var/obj/O = A
+			O.take_damage(erosionpower, BRUTE, "melee", 0)
+		else if(istype(A, /turf/closed))
+			return ..()
```
this code will never be reached because you early return if `A` is a turf
```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+		else if(istype(A, /turf/closed))
+			return ..()
+
+	return
```
erroneous return
```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+	SSaugury.unregister_doom(src)
+
+/obj/effect/meteor/atmospheric_drag/chase_target(atom/chasing, delay, home)
+	. = ..()
```
```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+
+/obj/effect/meteor/atmospheric_drag/Bump(atom/A)
+	// Pass through mobs harmlessly
+	if(isliving(A) || ismob(A))
```
```suggestion
	if(ismob(A))
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+		// Cache the bounds for this Z-level
+		if(found_station && max_x >= min_x && max_y >= min_y)
+			var/key = "[z_level]"
+			station_bounds_cache[key] = list("min_x" = min_x, "max_x" = max_x, "min_y" = min_y, "max_y" = max_y)
```
```suggestion
			station_bounds_cache["[z_level]"] = list("min_x" = min_x, "max_x" = max_x, "min_y" = min_y, "max_y" = max_y)
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+
+		// Iterate through all non-space areas
+		for(var/area/A in GLOB.areas)
+			if(A.type == /area/space || istype(A, /area/space))
```
```suggestion
			if(istype(A, /area/space))
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+SUBSYSTEM_DEF(orbital_visuals)
+	name = "Orbital Visuals"
+	can_fire = TRUE
+	wait = 2 // 0.2 seconds - fast enough for re-entry flicker, slow enough to not spam animate()
```
```suggestion
	wait = 0.2 SECONDS // fast enough for re-entry flicker, slow enough to not spam animate()
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+	/// Whether starlight control is overridden (like aurora caelus)
+	var/starlight_override = FALSE
+	/// Target color for starlight in current state
+	var/target_starlight_color = COLOR_STARLIGHT
```
unused var i think? github sometimes doesn't show me all occurrences of a variable when i ask for them
```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+
+	// Update all connected clients' orbital visual layers based on altitude
+	for(var/client/C in GLOB.clients)
+		if(!C?.mob?.hud_used)
```
`C` will never be null here
```suggestion
		if(!C.mob?.hud_used)
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_drag.dm:
```diff
@@ -0,0 +1,232 @@
+		can_fire = FALSE
+		return SS_INIT_SUCCESS
+
+	return SS_INIT_SUCCESS
```
`can_fire` already defaults to `FALSE`. add `SS_NO_INIT` to the subsystem flags

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_erosion.dm:
```diff
@@ -0,0 +1,206 @@
+	// Get reentry direction from map config
+	reentry_direction = SSmapping.current_map.reentry_direction
+
+	return SS_INIT_SUCCESS
```
`can_fire` already defaults to `FALSE`
```suggestion
	if(!SSmapping.current_map.planetary_station)
		// Get reentry direction from map config
		reentry_direction = SSmapping.current_map.reentry_direction
	return SS_INIT_SUCCESS
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_scanning.dm:
```diff
@@ -0,0 +1,217 @@
+/datum/controller/subsystem/orbital_reentry_scanning/Initialize()
+	// Planetary stations don't orbit — nothing to scan.
+	if(SSmapping.current_map.planetary_station)
+		can_fire = FALSE
```
`can_fire` already defaults to `FALSE`
```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_reentry_scanning.dm:
```diff
@@ -0,0 +1,217 @@
+	for(var/atom/movable/thing in tile)
+		if(isobj(thing) && !iseffect(thing))
+			return TRUE
+		if(ismob(thing) && !isobserver(thing))
```
this should probably only check for `isliving()`, there are a bunch of non-observer, non-living mobs. i.e: camera eyes, oranges ear, and another one i'm forgetting
```suggestion
		if(isliving(thing))
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+/datum/controller/subsystem/orbital_visuals/proc/update_normal_starlight()
+	// Optional: Tie to time of day by checking SSnatural_light_cycle
+	// For now, just maintain normal starlight
+	return
```
don't leave this here

```suggestion
```

mrmanlikesbt commented on code/controllers/subsystem/station_orbital/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+	var/reentry_depth = clamp((ORBITAL_ALTITUDE_LOW_CRITICAL - altitude) / (ORBITAL_ALTITUDE_LOW_CRITICAL - ORBITAL_ALTITUDE_LOW_BOUND), 0, 1)
+
+	// Violent flickering between different intensities of orange and red
+	var/list/reentry_colors = list(
+		"#FF4500", // Red-orange
+		"#FF6B00", // Orange-red
+		"#FF8C00", // Dark orange
+		"#FFA500", // Orange
+		"#FF0000", // Pure red (rare, intense moments)
+	)
```
make these color defines or use already existing ones in `__DEFINES\colors.dm`

("#FFA500" is used a ton across the codebase, it 100% needs its own define)
```suggestion
	var/static/list/reentry_colors = list(
		"#FF4500", // Red-orange
		"#FF6B00", // Orange-red
		"#FF8C00", // Dark orange
		"#FFA500", // Orange
		COLOR_RED, // Pure red (rare, intense moments)
	)
```

mrmanlikesbt commented on code/controllers/subsystem/orbital_visuals.dm:
```diff
@@ -0,0 +1,247 @@
+	g = clamp(g + ((255 - g) * amount), 0, 255)
+	b = clamp(b + ((255 - b) * amount), 0, 255)
+
+	return rgb(r, g, b)
```
remove this proc and use `BlendRGB(YOUR_COLOR_ARG, COLOR_WHITE, YOUR_AMOUNT_ARG)` instead. if that doesn't work the way you want it to, do the following:

move this to `__HELPERS\icons.dm` with the rest of the color proc helpers and make use of byond's `rgb2num()` proc

```suggestion
/// Helper proc to lighten a color
/proc/LightenRGB(color, amount)
	var/list/rgb = rgb2num(color)
	
	// Lighten by moving towards white (255)
	rgb[1] = clamp(rgb[1] + ((255 - rgb[1]) * amount), 0, 255)
	rgb[2] = clamp(rgb[2] + ((255 - rgb[2]) * amount), 0, 255)
	rgb[3] = clamp(rgb[3] + ((255 - rgb[3]) * amount), 0, 255)

	return rgb(rgb[1], rgb[2], rgb[3])
```

mrmanlikesbt commented on code/datums/map_config.dm:
```diff
@@ -213,3 +226,31 @@
+			if(converted)
+				log_runtime("ERROR: map_config loaded reentry_direction '[direction_text]' from file '[full_path]' (stale cache fallback).")
+				return converted
+	return null
```
don't do this. if the map config file is improperly configured, let it "break". "break" is in parentheses because  it'll retain the default `reentry_direction` var
```suggestion
```

mrmanlikesbt commented on code/datums/map_config.dm:
```diff
@@ -167,6 +167,19 @@
+		// Try to read the direction from the a _maps/ JSON for this map, just in case.
+		var/canonical_dir = load_reentry_direction_from_file_fallback(map_name)
+		if(canonical_dir)
+			reentry_direction = canonical_dir
```
see the other comment in this file
```suggestion
```

mrmanlikesbt commented on code/game/objects/structures/orbital_thruster.dm:
```diff
@@ -0,0 +1,311 @@
+	name = "orbital thruster"
+	desc = "A massive thruster used to adjust the station's orbit."
+	icon = 'icons/obj/orbital_thruster.dmi'
+	max_integrity = 50000000
```
does having `INDESTRUCTIBLE` in the `resistance_flags` not work?
```suggestion
```

TsunamiAnt replied:
better safe than sorry!

mrmanlikesbt commented on code/modules/atmospherics/gasmixtures/gas_types.dm:
```diff
@@ -220,6 +220,20 @@ GLOBAL_LIST_INIT(hardcoded_gases, list(/datum/gas/oxygen, /datum/gas/nitrogen, /
+	rarity = 100
+	base_value = 3
+	desc = "A specialized hydrogen-based fuel formulation designed for orbital maneuvering systems. Highly flammable and self-oxidizing. Keep away from open flames, this stuff will burn even in a vacuum."
+	primary_color = "#00ffff"
```
```suggestion
	primary_color = COLOR_CYAN
```

mrmanlikesbt commented on code/modules/shuttle/super_cruise/orbital_map_components/orbital_objects/space_station.dm:
```diff
@@ -17,7 +17,7 @@
 		mass = SSmapping.current_map.planet_mass
 		if (SSmapping.current_map.planet_name)
-			name = "[SSmapping.current_map.planet_name] (Outpost 13)"
+			name = "[SSmapping.current_map.planet_name]"
```
woke liberals removing the 13 from SS13
```suggestion
			name = "[SSmapping.current_map.planet_name] (Outpost 13)"
```

mrmanlikesbt commented on code/modules/shuttle/super_cruise/orbital_map_components/orbital_map.dm:
```diff
@@ -41,6 +41,16 @@
+	all_bodies = sortTim(all_bodies, GLOBAL_PROC_REF(cmp_orbital_priority_dsc))
+
+	// Now initialize in the correct order
+	for(var/datum/orbital_object/body as() in all_bodies)
```
```suggestion
	for(var/datum/orbital_object/body as anything in all_bodies)
```

mrmanlikesbt commented on code/game/objects/structures/orbital_thruster.dm:
```diff
@@ -0,0 +1,311 @@
+
+/obj/machinery/atmospherics/components/unary/orbital_thruster/process_atmos()
+	..()
+
```
don't parent call here
```suggestion
```

mrmanlikesbt commented on code/modules/lighting/starlight.dm:
```diff
@@ -1,6 +1,16 @@
 
 GLOBAL_VAR_INIT(starlight_colour, COLOR_STARLIGHT)
 
+GLOBAL_VAR_INIT(orbital_visual_starlight, COLOR_STARLIGHT)
+
+/// Sets the starlight colour and sends the signal. Used by anything that wants to change starlight colour globally.
+/// Override orbital visuals if you also want to touch station-Zs too.
 /proc/set_starlight_colour(new_colour, transition_time)
 	GLOB.starlight_colour = new_colour
 	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_STARLIGHT_COLOUR_CHANGE, new_colour, transition_time)
+
+/// Sets the orbital visuals starlight colour and sends the signal. Used by the orbital visual subsystem itself to update station Z-levels.
+/// Make SURE you tell it you are overriding stuff if you call this directly. See Aurora Caelus for example usage.
+/proc/set_orbital_starlight_colour(new_colour, transition_time)
+	GLOB.orbital_visual_starlight = new_colour
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_STARLIGHT_COLOUR_CHANGE, new_colour, transition_time)
```
so that i'm understanding correctly- `starlight_colour` is for the starlight of ALL space and `orbital_starlight_colour` is for station space only?

TsunamiAnt replied:
uhhh....

orbital_visual_starlight is the one we use on station turfs that adapts to orbital conditions.

starlight_color is the fallback for every other z, which is likely to mostly be the standard one unless something wacky is happening.

mrmanlikesbt commented on code/controllers/subsystem/processing/orbits.dm:
```diff
@@ -234,3 +237,16 @@ PROCESSING_SUBSYSTEM_DEF(orbits)
+	for(var/map_key in orbital_maps)
+		var/datum/orbital_map/orbital_map = orbital_maps[map_key]
+		for(var/zone in orbital_map.collision_zone_bodies)
+			for(var/datum/orbital_object/object as() in orbital_map.collision_zone_bodies[zone])
```
```suggestion
			for(var/datum/orbital_object/object as anything in orbital_map.collision_zone_bodies[zone])
```

mrmanlikesbt commented on code/_onclick/hud/orbital_visuals.dm:
```diff
@@ -0,0 +1,300 @@
+
+	// if the mob is not on a station Z, always assume the highest defined.
+	var/current_altitude = ORBITAL_ALTITUDE_HIGH_BOUND
+	if(screenmob.z in SSmapping.levels_by_trait(ZTRAIT_STATION))
```
bruh
```suggestion
	if(is_station_level(screenmob.z))
```

mrmanlikesbt reviewed (changes requested):
blocking requested changes not included in the comments:
 
- you use single letter var names everywhere, stop
- give attribution for the new sounds
- non-blocking as of now, but most of these new planet names sound stupid and i would hate if we ended up using them
- seeing as you're adding 6 orbital height related subsystems, they should all be in their own folder- if subsystems can be merged together, they should be
- the orbital control console should be in the CE's office, not the bridge

design questions:

i was under the impression the orbital thrusters would use plasma as laid out in the design doc, what made you shift away from this and add hydrogen fuel?

when i was going to pickup orbital thrusters myself (and you beat me to it) i originally thought about using the pre-existing ones and having them be in space, like you have now. but then i thought that it would be cooler if they were indoors and then pointing down. (enjoy my doodle)

<img width="374" height="400" alt="Image" src="https://github.com/user-attachments/assets/84a7d7ff-c95c-478d-9641-08e8915c0eaa" />

one last thing: the way you have orbital height set up now, you don't really interact with the thrusters at all. there are mainly 2 parts to the whole mechanic- fuel supply & and the control console. the way this exists now, in my opinion, the only interesting part about orbital height is the station being destroyed and the pretty UI.

one last one last thing: i worry fundamentally for orbital height and wonder if it's actually a good idea or if it was just something that sounded cool on paper during the creation of the design doc. in other words: do we think that introducing a new round-ending maintenance mechanic will actually improve the game? there are no benefits to keeping your orbital height high except for not ending the round. i feel like players will come to find this frustrating and unfun to play with. i would say we can see how this plays out on testmerge, but as i've learned, 90% of the time, players glaze a new mechanic when it's being testmerged because it's new and exciting, and only when it has become "solidified" in the game, do they form their real opinions.

TsunamiAnt committed:
Fix station tanks

TsunamiAnt committed:
Clean up atmos mapping helpers + adjust volume

TsunamiAnt committed:
typos

TsunamiAnt committed:
Better looking background

TsunamiAnt committed:
make agr use proper colors

TsunamiAnt committed:
Make selling plasma gas very profitable, as it should be.

TsunamiAnt committed:
Tutorial notes

TsunamiAnt committed:
No more single letter vars

TsunamiAnt committed:
Gasrig is now capable of experiencing pain and can also scream now

TsunamiAnt committed:
Slight adjustment to asteroid spawning to make it look less messy

TsunamiAnt committed:
Orbital thruster alerts/gas rig name fix

TsunamiAnt committed:
Update code/_onclick/hud/orbital_visuals.dm

Co-authored-by: mrmanlikesbt <99309552+mrmanlikesbt@users.noreply.github.com>

TsunamiAnt committed:
Update code/_onclick/hud/orbital_visuals.dm

Co-authored-by: mrmanlikesbt <99309552+mrmanlikesbt@users.noreply.github.com>

TsunamiAnt committed:
Use dimension-specific icon size constants for orbital visuals

Replaces world.icon_size with ICON_SIZE_X and ICON_SIZE_Y in orbital tile count calculations to ensure scaling remains accurate across both axes.

TsunamiAnt committed:
Remove unused afterShuttleMove area proc

Removes the empty afterShuttleMove proc and the associated loop in the shuttle docking cleanup logic, as the procedure performed no actions and had no overrides.

TsunamiAnt committed:
Remove unused performance configuration file

Deletes the empty config/performance.txt file, as it is no longer utilized by the system.

TsunamiAnt committed:
Move static orbital band data to ui_static_data

Relocates the constant orbital visualization bands from ui_data to ui_static_data. This optimizes UI performance by ensuring static data is only sent to the client once rather than on every periodic update.

TsunamiAnt committed:
Remove redundant return in orbital height control UI logic

Removes an unnecessary return statement in the orbital height control computer's UI action handling, as the return value is already assigned via the dot variable.

TsunamiAnt committed:
remove /obj/machinery/computer/orbital_height_control/ui_state(mob/user)

TsunamiAnt committed:
remove isliving from canPass drag

TsunamiAnt committed:
same for pass through

TsunamiAnt committed:
atmospheric_drag/Bump(atom/bumped) check code quality

TsunamiAnt committed:
remove dead branch in orbital_reentry_drag.dm

TsunamiAnt committed:
remove atmospheric_drag/chase_target

TsunamiAnt committed:
no need for isliving on atmospheric_drag/Bump

TsunamiAnt committed:
station_bounds_cache condensed to one line

TsunamiAnt committed:
orbital_reentry_drag.dm delete A.type == /area/space ||

TsunamiAnt committed:
use second define on orbital visuals wait

TsunamiAnt committed:
remove unused target_starlight_color

TsunamiAnt committed:
Client will never be null here

TsunamiAnt committed:
bunch of extra terrestrial checking nixed

TsunamiAnt committed:
bunch of little stuff

TsunamiAnt committed:
Remove lightenRGB and use blendRGB instead

TsunamiAnt committed:
Woops

TsunamiAnt committed:
rip load_reentry_direction_from_file_fallback, i was so proud of you...

TsunamiAnt committed:
More small stuff

TsunamiAnt committed:
extra return = gone

TsunamiAnt committed:
Very strange code. Code from another place.

TsunamiAnt committed:
Console + move stuff in folder

TsunamiAnt committed:
Update star layer blend mode to overlay

Changes the blend mode for the stars orbital layer from BLEND_ADD to BLEND_OVERLAY to improve visual rendering against the background.

TsunamiAnt committed:
stop them from yelling

TsunamiAnt committed:
Adjust orbital fire fade-out altitude threshold

Updates the fire fade-out logic to use a specific altitude value of 92000 instead of the ORBITAL_ALTITUDE_LOW constant to refine visual transitions.

TsunamiAnt committed:
Gateway integration

Adds operational constraints to station gateways based on the current orbital altitude. Gateways now require an altitude between 100km and 110km to align and maintain a stable tunnel, providing visual feedback and error messages when these thresholds are not met.

TsunamiAnt committed:
Cargo Integration

Links cargo shuttle efficiency to the station's orbital stability. When the station drops below 100km, shuttle flight times now increase linearly up to a 10x multiplier at 80km. Cargo consoles have been updated to display advisories and warnings when orbital deviation is affecting logistics.

TsunamiAnt committed:
Solar Power Integration

Links solar panel efficiency to the station's orbital altitude. Power output now scales linearly based on atmospheric interference, ranging from 0% at 100km to 200% at 140km. The solar control console has been updated with efficiency readouts and examine descriptions to reflect these conditions.

TsunamiAnt committed:
Consolidated Orbital Altitude Thresholds

Standardizes orbital altitude defines into a single hierarchy to be used across all related systems, including the Gateway, Cargo Shuttle, Solar Panels, and Re-entry Erosion. This replaces overlapping system-specific constants with a unified source of truth, ensuring consistent behavior and simplifying maintenance of altitude-based mechanics.

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
Orbital Area Path Refactor

Updates area type references within the radiation band and reentry drag subsystems to align with the current area hierarchy. This ensures that radiation protection levels and atmospheric drag exclusions are correctly applied to the intended station and space zones.

TsunamiAnt committed:
damn APC

Rukofamicom reviewed (dismissed):
Address and/or respond to the comments bacon left originally with a reason why you are not doing what is requested.

TsunamiAnt committed:
As anything / is_valid_hit

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
Clean up of a massive garbage patch of code

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
Take meta from master

TsunamiAnt committed:
Run smartwire fix

TsunamiAnt committed:
Implement orbital thruster infrastructure on MetaStation

Adds the necessary components for the orbital height system, including pink fuel supply lines, thruster nozzles, and the orbital height control console. Replaces localized gas-specific engine turfs with vacuum turfs and adjusts the reentry direction to South to align with the new map configuration.

TsunamiAnt committed:
goddamn

mrmanlikesbt reviewed (changes requested):
things i noticed on testmerge:
 
- orbital thrusters should burn turfs
- the "ideal" orbital background needs to have more flair. pure black with the occasional shooting star isn't cutting it
- set the AGR nozzle extension step size to 100

TsunamiAnt committed:
Overhaul space background

TsunamiAnt committed:
Same but icons

TsunamiAnt committed:
Fixes faulty gateway logic

TsunamiAnt committed:
Timestep from 1 to 0.5

TsunamiAnt committed:
Increase orbital thruster damage to turfs and objects

Thrusters now apply damage to the underlying turf and deal full damage to objects within the blast radius, ensuring consistent destruction of the affected area.

TsunamiAnt committed:
Fix orbital altitude flag transition logic

Ensures that the high altitude flag is set when entering critical altitude thresholds and decouples the clearing logic so that flags are reset independently as the station descends past their respective altitude markers.

TsunamiAnt committed:
Move orbital control logic to the subsystem to stop the consoles fightin

TsunamiAnt committed:
Metastation fully complete. Also map template for thruster

TsunamiAnt committed:
Increase Atmos Gas Rig extension step to 100

Adjusts the step increment for the gas rig extension input to allow for quicker adjustments over the larger distances required by orbital heights.

TsunamiAnt committed:
Meta fixes

TsunamiAnt committed:
Boxstation

TsunamiAnt committed:
MetaStation and RadStation orbital height updates

- Adds orbital thruster dummies and hydrogen fuel supply lines to MetaStation and RadStation.
- Standardizes atmospheric tank floors to vacuum.
- Cleans up various pipe layers and lattice structures for orbital height compatibility.

BriggsIDP committed:
Rad map update

TsunamiAnt committed:
Full system

TsunamiAnt committed:
way too many roids

TsunamiAnt committed:
Merge remote-tracking branch 'origin/orbitalheight' into orbitalheight

TsunamiAnt committed:
checkfix

TsunamiAnt committed:
smh don't use layer var

TsunamiAnt committed:
come on

BriggsIDP committed:
map changes

BriggsIDP committed:
Merge branch 'orbitalheight' of https://github.com/TsunamiAnt/BeeStation-Hornet into orbitalheight

TsunamiAnt committed:
Fix Card try 1

TsunamiAnt committed:
re-fix rad grrr

TsunamiAnt committed:
rad 4 electric boogaloo

TsunamiAnt committed:
delta?

TsunamiAnt committed:
fland attempt

TsunamiAnt committed:
Last checklists

TsunamiAnt committed:
Rad fix

TsunamiAnt committed:
Fland checklist

TsunamiAnt committed:
card checklist

BriggsIDP committed:
added railings back to rad :)

BriggsIDP committed:
Merge remote-tracking branch 'upstream/HEAD' into orbitalheight

TsunamiAnt committed:
Fix rad

TsunamiAnt committed:
Merge remote-tracking branch 'BeeStation/master' into orbitalheight

TsunamiAnt committed:
Orbital Thruster: Introduce configurable fuel parameters and refine comments

TsunamiAnt committed:
Orbital Thruster Nozzle: Refactor logic, effects, and damage system

TsunamiAnt committed:
Orbital Thruster: Overhaul fuel processing and unique naming scheme

TsunamiAnt committed:
Gateway fixes

TsunamiAnt committed:
Orbital Shuttle: Refine runaway recall mechanism

Update the shuttle runaway detection and recall system to improve robustness and player experience. Shuttles are now returned to a safe location near the station, mitigating accidental warps into hazardous areas like stars. The detection boundary for runaway shuttles now utilizes the configurable `ORBITAL_RUNAWAY_THRESHOLD`.

TsunamiAnt committed:
oop i forgot-  also fix typo

TsunamiAnt committed:
Increase roundstart air gas supply

TsunamiAnt committed:
Better note

TsunamiAnt committed:
merge master and deconflict maps

TsunamiAnt committed:
rerun please
